backends/xnnpack/runtime/XNNExecutor.cpp

```diff
@@ -106,20 +106,25 @@ ET_NODISCARD Error XNNExecutor::prepare_args(EValue** args) {
           err == Error::Ok,
           Internal,
           "Failed to retrieve dim order from tensor!");
-      ET_CHECK_OR_RETURN_ERROR(
-          is_contiguous_dim_order(dim_order, tensor->dim()),
-          Internal,
-          "Expecting default dim_order but got a non default dim_order tensor for external input %u",
-          i);
       size_t dims[XNN_MAX_TENSOR_DIMS];
       ET_CHECK_OR_RETURN_ERROR(
           num_dims <= XNN_MAX_TENSOR_DIMS,
           InvalidArgument,
           "XNNPACK backend accepts tensors with at most %d dims, but got %zu",
           XNN_MAX_TENSOR_DIMS,
           num_dims);
-      for (int d = 0; d < num_dims; ++d) {
-        dims[d] = tensor->size(d);
+
+      bool is_channels_last =
+          executorch::runtime::is_channels_last_dim_order(dim_order, num_dims);
+      if (is_channels_last) {
```

Comment:
can we just always convert? meaning something like:
```
for(int i = 0; i < dim_order.size(); ++i){
    dims[i] = dim_order[i];
}
```
if dim_order is 0, 1, 2, 3 then this is normal, otherwise channels last dim order is 0, 2, 3, 1